[Feat] Add info icon to selectors (#1124)
mckinsey/vizro: vizro-core/docs/pages/user-guides/selectors.md

    When configuring the [`DatePicker`][vizro.models.DatePicker] make sure to provide your dates for `min`, `max` and `value` arguments in `"yyyy-mm-dd"` format or as `datetime` type (for example, `datetime.datetime(2024, 01, 01)`).

## The `description` argument

> The `description` argument allows you to provide additional context for your selector. When `description` is defined, an icon appears to the right of the selector's title, and hovering over this icon reveals a tooltip containing the specified description text.
> 
> The `description` argument accepts either a plain `str`, in which case a default info icon is used, or a custom [`Tooltip`][vizro.models.Tooltip] model. The [`Tooltip`][vizro.models.Tooltip] model enables users to define both the tooltip content and a custom icon name from [Google Material icons library](https://fonts.google.com/icons).

Reviewer: Jo corrected me so often on this, so I am just passing it on 😄 We don't use "allow you" but always "enable you".

```suggestion
The `description` argument enables you to add helpful context to your selector by displaying an info icon next to its title. Hovering over the icon shows a tooltip with your provided text.

You can provide a string to use the default info icon, or pass a custom [`Tooltip`][vizro.models.Tooltip] model to define both a custom icon from the [Google Material Icons library](https://fonts.google.com/icons) and the description text.
```

> !!! example "Selectors with `description`"

Reviewer: ```suggestion
!!! example "Selectors with info-icon"
```

    === "app.py"

        ```{.python pycafe-link hl_lines="19"}
        import vizro.models as vm
        import vizro.plotly.express as px
        from vizro import Vizro

        iris = px.data.iris()

        page = vm.Page(
            title="Selectors with icons",
            components=[
                vm.Graph(
                    figure=px.scatter(iris, x="sepal_length", y="sepal_width")
                ),
            ],
            controls=[
                vm.Filter(
                    column="species",
                    selector=vm.Checklist(
                        title="Select Species",
                        description="""
                            Select which species of iris you like.

>                             [Click here](www.google.com) to learn more about flowers.""",

Reviewer 2: maybe even the origins of that data :)?

                    )
                ),

>                 vm.Filter(
>                     column="sepal_length",
>                     selector=vm.RangeSlider(
>                         title="Select sepal length",
>                         description=vm.Tooltip(
>                             text="Select species sepal_lenght range.",
>                             icon="help",
>                         ),
>                     )

Reviewer: Would simplify this example and just add one selector, I think that's fine :)
```suggestion
```
